upload/admin/controller/module/smaily_for_opencart.php

```diff
@@ -99,9 +103,40 @@ public function index() {
 
         // RSS feed text.
         $this->data['rss_feed_title'] = $this->language->get('rss_feed_title');
-        $this->data['rss_feed_text'] = $this->language->get('rss_feed_text');
-        $this->data['rss_feed_url'] = $this->config->get('config_url') . 'index.php?route=smailyforopencart/rss';
-
+        $this->data['rss_feed_text']  = $this->language->get('rss_feed_text');
+        $this->data['smaily_rss_url_base'] = $this->config->get('config_url') . 'index.php?route=smailyforopencart/rss&';
+        $rss_query_parameters = array();
+        $rss_settings = isset($this->model_setting_setting->getSetting('smaily')['smaily_rss'])
+            ? $this->model_setting_setting->getSetting('smaily')['smaily_rss'] : array();
```

Comment:
Since `$this->model_setting_setting->getSetting('smaily')` is queried before, I can actually drop the extra queries to 0 by using `$settings` and your implementation